app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true, length: {maximum: Settings.length.digit_50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i.freeze
+  validates :email, presence: true,
+            length: {maximum: Settings.length.digit_255},
+            format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  validates :password, presence: true,
+            length: {minimum: Settings.length.digit_6}
+  has_secure_password
```

Comment:
Hàm này có tác dụng gì em

Reply:
Dạ theo em tìm hiểu thì hàm:
  has_secure_password(attribute = :password, validation: true) có ba chức năng :
       - Cung cấp khả năng lưu thuộc tính password_digest bằng mã băm trên db
       - Tạo cặp thuộc tính ảo chỉ có trên model là password và password_confirmmation, validation cho chúng ở mức không được thiếu (empty?) nên có trường hợp chỉ toàn khoảng trắng validate vẫn là true ( ta nên dùng validate presence: true để tránh trường hợp này)
       - Phương thức authenticate trong nó sẽ trả về user khi xác thực đúng và ngược lại là false.